FPEngine.cc

```diff
@@ -479,6 +490,25 @@ void FPNetworkControl::probe_transmission_handler(nsock_pool nsp, nsock_event ns
         myprobe->host->fail_one_probe();
         gh_perror("Unable to send packet in %s", __func__);
       }
+      /* If Packet was sent successfully and Decoys are present */
+      else if (o.numdecoys > 1) {  
+        FPPacket *temp = new FPPacket();
+        std::map<std::string, IPv6Header*>::iterator it;
+        char addr[INET6_ADDRSTRLEN];
+        for (int decoy = 0; decoy < o.numdecoys; decoy++) {
+          if (decoy == o.decoyturn)
+            continue;
```

Comment:
@dmiller-nmap I agree with your idea and I think that having a function to swap out address from header would be the best option as far as efficiency is concerned. One more small fix, though not that much good, could be -
Iterate over decoys like we are doing now but on finding index as `decoyturn` we would send `(FPProbe *)arg` to `send_ip_packet()` and setting the time sent .
If we do something like sending an array of `FPPacket` or `FPProbe` then it would be more or less similar to the above idea except the fact that original packet as well as decoys will come together to the function.
Looking forward to your thoughts on this.